### PR TITLE
Benji/ios 571 emotion selection

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/UI/Transitions/DismissTransitionController.swift [hunk 1]
```diff
@@ -45,57 +45,67 @@ class DismissTransitionController: NSObject, UIViewControllerAnimatedTransitioni
         guard self.animator.isNil else {
             return
         }
+        
+        var toVC: MessageInteractableController?
```
Comment (Benjamindodgson) [inline Shared/UI/Transitions/DismissTransitionController.swift:49]: More generic dismissal

File: iOS/Flows/Conversation/Input/AddMediaView.swift [hunk 2]
```diff
@@ -37,6 +37,7 @@ class AddMediaView: ThemeButton {
         self.layer.cornerRadius = Theme.innerCornerRadius
                 
         self.addSubview(self.displayableImageView)
+        self.showsMenuAsPrimaryAction = true
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/AddMediaView.swift:40]: No long press to delete needed.

File: iOS/Flows/Message Detail/MessageDetailViewController.swift [hunk 3]
```diff
@@ -96,20 +96,26 @@ class MessageDetailViewController: DiffableCollectionViewController<MessageDetai
...
             return !reaction.author.isCurrentUser
         }).compactMap({ read in
-            return .read(read)
+            return .read(ReadViewModel(readReaction: read))
         })
         
-        if msg.replyCount > 0 {
-            data[.recentReply] = [.reply(msg)]
+        if reads.isEmpty {
+            data[.reads] = [.read(ReadViewModel(readReaction: nil))]
```
Comment (Benjamindodgson) [inline iOS/Flows/Message Detail/MessageDetailViewController.swift:112]: I have to look into why the section background is being displayed, even though not items are in it, so for now I add an empty cell.

File: iOS/Flows/Message Detail/RecentReplyView.swift [hunk 4]
```diff
@@ -57,35 +67,56 @@ class RecentReplyView: CollectionViewManagerCell, ManageableCell {
...
+        self.label.isVisible = false
     }
     
     func configure(with item: Message) {
         Task.onMainActorAsync {
             
             if self.controller?.message != item {
                 self.controller = ChatClient.shared.messageController(cid: item.cid!, messageId: item.id)
-
+                self.animationView.play()
```
Comment (Benjamindodgson) [inline iOS/Flows/Message Detail/RecentReplyView.swift:82]: Shows a loading state for a recent reply.

File: iOS/Flows/Threads/ThreadViewController.swift [hunk 5]
```diff
@@ -399,6 +421,12 @@ extension ThreadViewController: TransitionableViewController {
     func handleDismissal() {
         self.pullView.bottom = self.messageContent.top
     }
+    
+    func handleCompletedDismissal() {
+        if let selectedIndexPath = self.collectionView.indexPathsForSelectedItems?.first {
```
Comment (Benjamindodgson) [inline iOS/Flows/Threads/ThreadViewController.swift:408]: Make sure we deselect, so we don't dismiss the wrong message content.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**